src/CMakeLists.txt

```diff
@@ -700,6 +700,10 @@ endif(WITH_CCACHE AND CCACHE_FOUND)
 list(APPEND ROCKSDB_CMAKE_ARGS -DCMAKE_AR=${CMAKE_AR})
 list(APPEND ROCKSDB_CMAKE_ARGS -DCMAKE_BUILD_TYPE=${CMAKE_BUILD_TYPE})
 
+if (CMAKE_CXX_COMPILER_ID STREQUAL Clang)
+  list(APPEND ROCKSDB_CMAKE_ARGS -DFAIL_ON_WARNINGS=OFF)
```

Comment:
fwiw, the patch silence the warning is posted at https://github.com/facebook/rocksdb/pull/1757.